Templates/AppSource App/.github/workflows/UpdateGitHubGoSystemFiles.yaml

```diff
@@ -74,18 +74,44 @@ jobs:
 
       - name: Calculate Input
         env:
-          directCommit: '${{ github.event.inputs.directCommit }}'
+          directCommit: ${{ github.event.inputs.directCommit }}
           downloadLatest: ${{ github.event.inputs.downloadLatest }}
           eventName: ${{ github.event_name }}
+          GhTokenWorkflow: ${{ fromJson(steps.ReadSecrets.outputs.Secrets).ghTokenWorkflow }}
         run: |
-          $errorActionPreference = "Stop"; $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
+          $errorActionPreference = "Stop";
+          $ProgressPreference = "SilentlyContinue"; Set-StrictMode -Version 2.0
           $directCommit = $ENV:directCommit
           $downloadLatest = $ENV:downloadLatest
           Write-Host $ENV:eventName
           if ($ENV:eventName -eq 'schedule') {
-            Write-Host "Running Update AL-Go System Files on a schedule. Setting DirectCommit and DownloadLatest to true"
-            $directCommit = 'true'
             $downloadLatest = 'true'
+            $directCommit = 'true'
+            try {
+              Write-Host "Running Update AL-Go System Files on a schedule."
+              $ENV:GH_TOKEN = [System.Text.Encoding]::UTF8.GetString([System.Convert]::FromBase64String($env:GhTokenWorkflow))
+              $rules = gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/$env:GITHUB_REPOSITORY/rules/branches/$env:GITHUB_REF_NAME | ConvertFrom-Json
+              $branchProtection = gh api -H "Accept: application/vnd.github+json" -H "X-GitHub-Api-Version: 2022-11-28" /repos/$env:GITHUB_REPOSITORY/branches/$env:GITHUB_REF_NAME/protection | ConvertFrom-Json
+              if ($rules | Where-Object { $_.type -eq 'pull_request' }) {
+                Write-Host "Pull request rule exists for branch $($env:GITHUB_REF_NAME)"
+                $directCommit = 'false'
+              }
+              elseif ($branchProtection.PSObject.Properties.Name -eq "required_pull_request_reviews") {
+                Write-Host "Branch protection requires pull request for branch $($env:GITHUB_REF_NAME)"
+                if ($branchProtection.enforce_admins.enabled) {
```

Comment:
Alternatively, instead of checking if rulesets and branch protection exist could we try to push to the branch (in CheckForUpdate.ps1) and just fall back to creating a PR if that fails?

Reply:
Implemented this idea.